In [1]:
from llm_adapter import RAG_Chroma

2024-07-09 16:06:42,721 - modelscope - INFO - PyTorch version 2.1.0 Found.
2024-07-09 16:06:42,722 - modelscope - INFO - Loading ast index from /data/usr/jy/asset/modelscope/hub/ast_indexer
2024-07-09 16:06:42,837 - modelscope - INFO - Loading done! Current index file version is 1.9.2, with md5 0e964997b9816d332d6c9290f1e202b9 and a total number of 941 components indexed
/home/jy/anaconda3/envs/baichuan2/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
rag=RAG_Chroma()

2024-07-09 16:06:45,906 - INFO - Load pretrained SentenceTransformer: /data/usr/jy/asset/tokenizer/m3e-base


2024-07-09 16:06:46,905 - INFO - Use pytorch device: cuda
2024-07-09 16:06:47,147 - DEBUG - Registering provider: token_config
2024-07-09 16:06:47,149 - DEBUG - Registering provider: user_token_config
2024-07-09 16:06:47,151 - DEBUG - Registering provider: token
2024-07-09 16:06:47,152 - DEBUG - Registering provider: token
2024-07-09 16:06:47,161 - INFO - Anonymized telemetry enabled. See                     https://docs.trychroma.com/telemetry for more information.
2024-07-09 16:06:47,250 - DEBUG - Starting component System
2024-07-09 16:06:47,251 - DEBUG - Starting component Posthog
2024-07-09 16:06:47,251 - DEBUG - Starting component OpenTelemetryClient
2024-07-09 16:06:47,252 - DEBUG - Starting component SimpleAssignmentPolicy
2024-07-09 16:06:47,252 - DEBUG - Starting component SqliteDB
2024-07-09 16:06:47,254 - DEBUG - Starting component LocalSegmentManager
2024-07-09 16:06:47,254 - DEBUG - Starting component SegmentAPI


1. 雷诺发现了管路中液体流动时具有哪两种状态    层流，紊流
> 英国物理学家雷诺通过大量实验，发现了液体在管路中流动时存在的两种流动状态--层流和紊流。

2. 影响磷酸酯抗燃液压油使用性能和寿命的主要因素有哪些    水解劣化，热裂解劣化及氧化劣化 （水分，高温及氧气）
> 由于磷酸酯分子的特殊性，在使用过程中的水分，高温及氧气均会导致磷酸酯抗燃液压油劣化。为此对磷酸酯抗燃液压油的发展和劣化（水解劣化，热裂解劣化及氧化劣化）机制进行了综述

3. 航空器上的液压设备为什么要选择闭式回路啊   为减小体积和质量
> 对于行走机械和航空航天液压装置，为减小体积和质量，可选择闭式回路，即执行元件的排油直接进入液压泵的进口。

4. 零下一百二十摄氏度的环境下使用O形密封圈，应该选用橡胶，聚四氟乙烯，还是空心金属材质的     空心金属
>  (1)聚四氟乙烯O形密封圈它比橡胶O形密封圈耐寒、耐热，其使用温度范围为-100～130℃   2)空心金属O形密封圈空心金属O形密封圈是用薄壁不锈钢管或镍合金管焊接而成的高精度环形圈。其密封特点是：1) 适用温度范围广(-250～540℃)。

5. NAS 1638固体颗粒污染物按照微粒大小分成了那五个等级
> NAS 1638油液固体颗粒污染度等级本标准将油液中的颗粒按尺寸大小分为5~15µm、15~25µm、25~50µm、50~100µm和大于100µm五个尺寸段

6. 举一个浙大研究液压的网站
> 2、有关网站中国液压气动网（www.yeyanet.com.cn) 浙江大学国家电液控制工程技术研究中心(www.cnehc.com)

7. Q2-8液压起重机的特点
> Q2－8型汽车起重机的液压系统有如下几个特点。（1）该系统为单泵、开式、串联系统，采用了换向阀串联组合，不仅各机构的动作可以独立进行，而且在轻载作业时，可实现起升和回转复合动作，以提高工作效率。（2）系统中采用了平衡回路、锁紧回路和制动回路，保证了起重机的工作可靠，操作安全。（3）采用了三位四通手动换向阀换向，不仅可以灵活方便地控制换向动作，还可通过手柄操纵来控制流量，实现节流调速。在起升工作中，将此节流调速方法与控制发动机转速的方法结合使用，可以实现各工作部件微速动作。（4）各三位四通手动换向阀均采用了M型中位机能，使换向阀处于中位时能使系统卸荷，可减少系统的功率损失，适宜于起重机进行间歇性工作

8. 什么是液压振动
> 液压振动指利用液体的压力能在动力系统的供压和伺服阀的控制下实现振动台的上下正玄运动。

9. 世界上第一台水压机是由谁在哪里发明的
> 1795年英国约瑟夫·布拉曼(Joseph Braman,1749-1814)，在伦敦用水作为工作介质,以水压机的形式将其应用于工业上,诞生了世界上第一台水压机。

10. 根据柯兵的研究成果，怎么样减少泵的压力脉动
> 柯兵等[33]对柱塞泵的压力脉动抑制方法进行了研究，在系统安装缓冲瓶的基础上，对柱塞目对液压泵的压力脉动的衰减效果进行了试验分析，结果表明，在定转速变流量和定流量变转速条件下，柱塞数为9和11的液压泵最大压力脉动分别为0.3 MPa、1.95 MPa和1.28 MPa、3.3 MPa。因此，选择合理的柱塞数配合缓冲瓶可抑制泵的压力脉动。

MIKOTA，欧阳小平和陈杰设计的脉动衰减器有哪些相同和不同之处"

In [10]:
query="液压泵有哪些类型的能量损失"
rag_docs=rag.query(query, top_k=20)
for r in rag_docs:
    print(len(r.page_content), "字", r.page_content, end="\n\n")

Batches: 100%|██████████| 1/1 [00:00<00:00, 65.03it/s]

442 字 : 有一点机械常识的人都知道，能量会互相转换的，而把这个知识运用到液压系统上解释液压系统的功率损失是最好不过了，液压系统功率一方面会造成能量上的损失，使系统的总效率下降，另一方面，损失掉的这一部分能量将会转变成热能，使液压油的温度升高，油液变质，导致液压设备出现故障。因此，设计液压系统时，在满足使用要求的前提下，还应充分考虑降低系统的功率损失。第一，从动力源——泵的方面来考虑，考虑到执行器工作状况的多样化，有时系统需要大流量，低压力；有时又需要小流量，高压力。所以选择限压式变量泵为宜，因为这种类型的泵的流量随系统压力的变化而变化。当系统压力降低时，流量比较大，能满足执行器的快速行程。当系统压力提高时流量又相应减小，能满足执行器的工作行程。这样既能满足执行器的工作要求，又能使功率的消耗比较合理。第二，液压油流经各类液压阀时不可避免的存在着压力损失和流量损失，这一部分的能量损失在全部能量损失中占有较大的比重。因此，合理选择液压器，调整压力阀的压力也是降低功率损失的一个重要方面

432 字 :
式中C———流量系数;
A———阀口通流面积, m2;
Δp———阀口先后压差, MPa;
ρ———流体密度, kg/m3。管路沿程压力损失公式为式 (6) 、 (7) 、 (8) 。式 (6) ~ (9) 可知, 流量对液压元件﹑管路压力损失影响极大。几大进口液压品牌样本都有液压元件流量压力损失曲线图, 设计者只需按通过液压元件的流量就可查得相应的压力损失, 这里须注意不能按系统流量去查。当前国产液压品牌样本鲜有液压元件流量压力损失曲线图, 希望完善样本。管路压力损失按式 (6) 、 (7) 、 (8) 计算后累加即可得出管路总压力损失, 可参看文献[6]。
3) 中位机能严重影响换向阀的压力损失电磁换向阀中位机能繁多, 相同流量下对应的压力损失相差很大 (参看Yuken、Rexroth样本) , 选型时须注意。
4) 液压元件﹑管道通径不能按系统流量选取因n一般为1.33~5, 甚至更大, 故无杆腔回油时经过液压元件﹑管道的流量就是系统流量的1

498 字 : 流量损失: 在液压系统中，各被压元件都有相对运动的表面，如液压缸内表面和活塞外表面，因为要有相对运动，所以它们之间都有一定的间隙。如果间隙的一边为高压油，另一边为低压油，则高压油就会经间隙流向低压区从而造

In [8]:
help(RAG_Chroma)

Help on class RAG_Chroma in module llm_adapter:

class RAG_Chroma(builtins.object)
 |  RAG_Chroma(db_path='./chroma_db/yeya', embedding_path='/data/usr/jy/asset/tokenizer/m3e-base') -> None
 |  
 |  Methods defined here:
 |  
 |  __init__(self, db_path='./chroma_db/yeya', embedding_path='/data/usr/jy/asset/tokenizer/m3e-base') -> None
 |      Initialize self.  See help(type(self)) for accurate signature.
 |  
 |  create(self, file_path: str = 'yeya-text12456.txt', chunk_size=500, chunk_overlap=50)
 |  
 |  delete(self)
 |  
 |  query(self, query: str)
 |  
 |  ----------------------------------------------------------------------
 |  Data descriptors defined here:
 |  
 |  __dict__
 |      dictionary for instance variables (if defined)
 |  
 |  __weakref__
 |      list of weak references to the object (if defined)

